### ISSUE TITLE
Bug with field chart and date range?

### ISSUE THREAD

**Issue by wazeefah-formtools** [formtools/module-data_visualization#23]
Hi

I have FormTools 3.0.15 and Data Visualisation 2.1.2.

I have created a field chart visualization for a form.

In the MAIN tab, I have chosen 'Date' as Field.
![Capture1](https://user-images.githubusercontent.com/56290806/66456754-897e1b00-ea7f-11e9-9ecc-43b3db9b8b5a.PNG)

In the APPEARANCE tab, I have chosen the following:
Chart Type: Column Chart
Date Range to display: Last 7 days
![Capture2](https://user-images.githubusercontent.com/56290806/66456755-8a16b180-ea7f-11e9-8eb3-0162d9ec5554.PNG)

The field chart is being displayed by taking ALL THE SUBMISSIONS from the form. It is not displaying according the date being selected.

Same applies for the other chart types and date range selected for the field charts.

However, for the activity charts, the charts display accordingly based on the date range selected, I was thinking that the field chart should work in the same way.

Can you please confirm if it's actually a bug? If not, can you please guide me how I should proceed to make the field charts display as per the date range I am selecting?

Thank you in advance.

**Comment by benkeen** [comment]
Hi @wazeefah-formtools, thanks for letting me know. Yes, this certainly looks like a bug. Let me just check it out and get back to you.

**Comment by benkeen** [comment]
I checked the code and it actually looks okay. Could you try running this command on your MySQL server? 
```
SELECT CURRENT_TIMESTAMP;
```

Looking at the code, the only likely cause of this problem is that your database server has a incorrect current datetime set. Seems unlikely, but running that command would settle the question.

**Comment by wazeefah-formtools** [comment]
Hi

The current timestamp seems to be okay. I am in Mauritius and thus using time zone (UTC + 04:00).

![Capture3](https://user-images.githubusercontent.com/56290806/66539223-5d72a080-eb38-11e9-93b4-a37c3893a257.PNG)

Please see the date field structure and some sample dates from the table.

![Capture4](https://user-images.githubusercontent.com/56290806/66539383-be01dd80-eb38-11e9-8684-760facb027b6.PNG)

![Capture5](https://user-images.githubusercontent.com/56290806/66539384-be01dd80-eb38-11e9-98b2-46be73f65caa.PNG)

I am having this issue only with the FIELD CHARTS when I am filtering on a field DATE which I have created. Even if I try to filter based on SUBMISSION DATE, it still does not work as in changing the charts dynamically.

But when choosing the ACTIVITY CHARTS, then everything works fine with the different options of data range and charts are updated accordingly - and that is based on the SUBMISSION DATE as well.

Please help.

**Comment by benkeen** [comment]
Oh! I didn't realize it was a custom date field, that makes a lot more sense.

What the "Date range to display" field does is _always_ tie in with the Submission Date field in submissions date. In other words, it'll display any data that's been added within the actual submission date range specified. But in your case, the data being *displayed* is your custom date field, so it'll show whatever data is in that field, within the date range (last 7 days). 

> Even if I try to filter based on SUBMISSION DATE, it still does not work as in changing the charts dynamically.

I don't quite follow what you mean here. You're saying it's not dynamically updating the data? But the Field is on a separate tab. So when you select the Submission Date field on the first tab, then go to the second tab the graph doesn't look correct either? Sorry not quite following.

**Comment by wazeefah-formtools** [comment]
Sorry, I was not clear enough. I will try to explain again.

There are two types of charts: activity charts and field charts.

For the activity charts, I don't have to select any fields, the graphs are displayed based on the submission date, this is working perfectly fine. 

My issue was with the field charts, when I am selecting a custom field ACCIDENT DATE in the tab MAIN and then I am moving to the tab APPEARANCE and choosing date range of last 7 days, the graphs are not displayed correctly based on the date range I am selecting. It is actually displaying all the accident dates of the form.

![Capture6](https://user-images.githubusercontent.com/56290806/66542150-cd862400-eb42-11e9-8e13-d77c59627041.PNG)
![Capture7](https://user-images.githubusercontent.com/56290806/66542151-cd862400-eb42-11e9-9561-9e0ef473f79b.PNG)

Then I was trying to understand why it is not working and I tried the following just to check if there was an issue with my custom field 'Accident date': 

I changed the field from the MAIN tab to DATE (Submission Date), and then moving to tab APPEARANCE and choosing last 7 days for date range, the chart is not displaying correctly, as in I was expecting it to display the last 7 submission dates. But again, it was taking all the submissions of the form.

![Capture8](https://user-images.githubusercontent.com/56290806/66542243-1b029100-eb43-11e9-8dca-22c1efbdb469.PNG)
![Capture9](https://user-images.githubusercontent.com/56290806/66542244-1b029100-eb43-11e9-8369-fc69cceab35f.PNG)

**Comment by wazeefah-formtools** [comment]
Actually I understood where the issue is coming. I saw in the code in the php file FieldCharts.class.php on lines 141- 154 that it's actually filtering the date based on the 'submission_date'. 

But I actually wanted to filter based on my custom field which I have named as 'accident_date', so I have changed the variable name in the codes from 'submission_date' to 'accident_date' and its working fine now.

Thanks a lot.
